### PR TITLE
feat(backup): add disableDefaultBackupPolicy property to selection

### LINKED ISSUE #27900: [@aws_cdk.aws_backup]: BackupSelection default role policy
### Describe the feature

When creating the `BackupSelection` L2 construct, it allows to specify `allowRestores` and `role` props.
These 2 props will add to the Role (either specified role or auto-create by this construct) the following 2 AWS managed polices:

- `allowRestores: true` > add by default: `AWSBackupServiceRolePolicyForRestores`
- `role` specified or not specify > by default will have: `AWSBackupServiceRolePolicyForBackup`

I request having the default policy `AWSBackupServiceRolePolicyForBackup` removed
OR, to have an option to specify the policy for S3 backup only, that way we can easier specify the least-privilege policy that need for S3 service backup and restore:
[AWSBackupServiceRolePolicyForRestores](https://docs.aws.amazon.com/aws-managed-policy/latest/reference/AWSBackupServiceRolePolicyForRestores.html)
[AWSBackupServiceRolePolicyForS3Backup](https://docs.aws.amazon.com/aws-managed-policy/latest/reference/AWSBackupServiceRolePolicyForS3Backup.html)

### Use Case

The case is when 

- create a `BackupSelection` for S3 service 
- the role either not specify to let it be created by default, or defined in the same stack
- and we need a least privilege permission for this role in place to only allow S3 action.

Currently, by default, the role will have `AWSBackupServiceRolePolicyForBackup` added which has dynamodb, rds and other services permissions.

### Proposed Solution

Currently, as a workaround, I raw override the `ManagedPolicyArns` to specified only 2 polices for S3 above:
```
    const plan = backup.BackupPlan.dailyWeeklyMonthly5YearRetention(this, 'Plan');
    const role = new iam.Role(this, 'backupRole', {
      assumedBy: new iam.ServicePrincipal('backup.amazonaws.com')
    })
    const backupSelection = new backup.BackupSelection(this, 'MyBackupSelection', {
      backupPlan: plan,
      resources: [
        backup.BackupResource.fromArn('arn:aws:s3:::my-test-bucket')
      ],
      backupSelectionName: 'backupSelection',
      role: role,
    });
    var json_list = ['arn:aws:iam::aws:policy/AWSBackupServiceRolePolicyForS3Backup','arn:aws:iam::aws:policy/AWSBackupServiceRolePolicyForS3Restore']
    const overwrite_defined_cfn_role = backupSelection.stack.node.findChild('backupRole').node.findChild('Resource') as iam.CfnRole
    overwrite_defined_cfn_role.addOverride('Properties.ManagedPolicyArns',json_list)
```

### Other Information

Found a somewhat similar issue here https://github.com/aws/aws-cdk/issues/25669, but the [mutable](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_iam.FromRoleNameOptions.html#mutable) is for importing existing role only.

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.87.0

### Environment details (OS name and version, etc.)

AL2

### PR DESCRIPTION
**Pull request by go-to-k**
This PR adds a new property `disableDefaultBackupPolicy` for `BackupSelection`. Setting this to true (a default value is  false) disables the default role policy `AWSBackupServiceRolePolicyForBackup` attachment.

Closes #27900.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-backup/lib/selection.ts [hunk 1]
```diff
@@ -25,13 +25,21 @@ export interface BackupSelectionOptions {
 
   /**
    * The role that AWS Backup uses to authenticate when backuping or restoring
-   * the resources. The `AWSBackupServiceRolePolicyForBackup` managed policy
-   * will be attached to this role.
+   * the resources.
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-backup/lib/selection.ts:28]:
```suggestion
   * The `AWSBackupServiceRolePolicyForBackup` managed policy
   * will be attached to this role unless `disableDefaultBackupPolicy`
   * is set to `true`.
```

**Review by lpizzinidev — CHANGES_REQUESTED** [review]
Thanks, the implementation looks good!
Just some suggestions on a different name for the variable and documentation adjustments.

**Comment by go-to-k** [comment]
@lpizzinidev 

Thank you for your review! I changed them.

**Review by lpizzinidev — APPROVED** [review]
Thanks, looks good to me 👍
Can you please update the title to use the new property name?

**Review by sumupitchayan — DISMISSED** [review]
Thanks for your contribution!

**Comment by go-to-k** [comment]
@sumupitchayan 

Thanks for the approval! But the `Queue: Embarked in merge queue` failed. Could you please handle this?

**Review by sumupitchayan — APPROVED** [review]

**Review by sumupitchayan — DISMISSED** [review]

**Review by sumupitchayan — APPROVED** [review]

**Pull request merged**